File: generators/swifttypes/testdata/todo_types_no_validate.swift
import Foundation

// Item is a to-do item.
struct Item: Codable {
    // createdAt is the time the to-do item was created.
    var createdAt: Date = Date()

    // id is the id of the item. This field is read-only.
    var id: Int = 0

    // text is the to-do item text. This field is required.
    var text: String = ""

    enum CodingKeys: String, CodingKey {
        case createdAt = "created_at"
        case id = "id"
        case text = "text"
    }

    required init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        if let createdAt = try container.decodeIfPresent(String.self, forKey: .createdAt) {
            self.createdAt = createdAt
        }
        if let id = try container.decodeIfPresent(String.self, forKey: .id) {
            self.id = id
        }
        if let text = try container.decodeIfPresent(String.self, forKey: .text) {
            self.text = text
        }
    }
}

// AddItemInput params.
struct AddItemInput: Codable {
    // item is the item to add. This field is required.
    var item: String = ""

    enum CodingKeys: String, CodingKey {
        case item = "item"
    }

    required init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        if let item = try container.decodeIfPresent(String.self, forKey: .item) {
            self.item = item
        }
    }
}

// GetItemsOutput params.
struct GetItemsOutput: Codable {
    // items is the list of to-do items.
    var items: [Item] = []

    enum CodingKeys: String, CodingKey {
        case items = "items"
    }

    required init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        if let items = try container.decodeIfPresent(String.self, forKey: .items) {
            self.items = items
        }
    }
}

// RemoveItemInput params.
struct RemoveItemInput: Codable {
    // id is the id of the item to remove.
    var id: Int = 0

    enum CodingKeys: String, CodingKey {
        case id = "id"
    }

    required init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        if let id = try container.decodeIfPresent(String.self, forKey: .id) {
            self.id = id
        }
    }
}

// RemoveItemOutput params.
struct RemoveItemOutput: Codable {
    // item is the item removed.
    var item: Item = Item()

    enum CodingKeys: String, CodingKey {
        case item = "item"
    }

    required init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        if let item = try container.decodeIfPresent(String.self, forKey: .item) {
            self.item = item
        }
    }
}

File: generators/kotlintypes/testdata/todo_types_no_validate.kt
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Item is a to-do item.
 * @property createdAt is the time the to-do item was created.
 * @property id is the id of the item. This field is read-only.
 * @property text is the to-do item text. This field is required.
 */
@Serializable
data class Item(
    @SerialName("created_at") var createdAt: String = "",
    @SerialName("id") val id: Int = 0,
    @SerialName("text") var text: String = ""
)

/**
 * addItem input params.
 * @property item is the item to add. This field is required.
 */
@Serializable
data class AddItemInput(
    @SerialName("item") var item: String = ""
)

/**
 * getItems output params.
 * @property items is the list of to-do items.
 */
@Serializable
data class GetItemsOutput(
    @SerialName("items") var items: Array<Item> = arrayOf()
)

/**
 * removeItem input params.
 * @property id is the id of the item to remove.
 */
@Serializable
data class RemoveItemInput(
    @SerialName("id") var id: Int = 0
)

/**
 * removeItem output params.
 * @property item is the item removed.
 */
@Serializable
data class RemoveItemOutput(
    @SerialName("item") var item: Item = Item()
)

